### PR TITLE
More flexible native6 default DRS

### LINKED ISSUE #641: Simplify default DRS for native6 project
Because the data used with the `native6` project can have any filename, it might be nice to make the default DRS
```
  input_dir:
    default: '{dataset}/{latestversion}/{frequency}/{short_name}'
  input_file:
    default: '*.nc'
```
because this would allow using the filenames that the data comes with without renaming. You could just download the data, put it in the right directory and be done. It would also avoid the issue reported here: https://github.com/ESMValGroup/ESMValTool/pull/1644#issue-413480838.

### PR DESCRIPTION
**Pull request by bouweandela**
Before you start, please read [CONTRIBUTING.md](https://github.com/ESMValGroup/ESMValCore/blob/master/CONTRIBUTING.md).

**Tasks**

-   [x] [Create an issue](https://github.com/ESMValGroup/ESMValCore/issues) to discuss what you are going to do, if you haven't done so already (and add the link at the bottom)
-   [x] This pull request has a descriptive title that can be used in a changelog
-   [x] Circle/CI tests pass. Status can be seen below your pull request. If the tests are failing, click the link to find out why.
-   [x] Codacy code quality checks pass. Status can be seen below your pull request. If there is an error, click the link to find out why. If you suspect Codacy may be wrong, please ask by commenting.
-   [x] Please use `yamllint` to check that your YAML files do not contain mistakes

If you need help with any of the tasks above, please do not hesitate to ask by commenting in the issue or pull request.

* * *

Closes #641

### CODE REVIEW

File: esmvalcore/config-developer.yml [hunk 1]
```diff
@@ -191,10 +191,10 @@ OBS6:
 native6:
   cmor_strict: false
   input_dir:
-    default: 'Tier{tier}/{dataset}'
+    default: '{dataset}/{latestversion}/{frequency}/{short_name}'
```
Comment (mattiarighi) [inline esmvalcore/config-developer.yml:194]: Why do we need `{latestversion}` here?

Comment (mattiarighi) [reply esmvalcore/config-developer.yml:194]: Also, wouldn't be more straightforward to use `era5_name` instead of `short_name`?

Comment (bouweandela) [reply esmvalcore/config-developer.yml:194]: > Why do we need {latestversion} here?

Because we want to support multiple versions of datasets

> Also, wouldn't be more straightforward to use `era5_name` instead of `short_name`?

Using the short name has the advantage that you can also use this DRS for other observational datasets than ERA5, making it easier to add support for those.

Comment (mattiarighi) [reply esmvalcore/config-developer.yml:194]: OK. I think I understand why it didn't work before.
Let me try again.

Comment (bouweandela) [reply esmvalcore/config-developer.yml:194]: I provided an example of how to use it in the issue (#641), maybe that helps?

Comment (mattiarighi) [reply esmvalcore/config-developer.yml:194]: Can we add `Tier{tier}/`, I think it would be good to keep this information in the directory tree.

Comment (bouweandela) [reply esmvalcore/config-developer.yml:194]: Do we really need those Tier1, 2, 3 directories? Is it possible for one dataset to be in two different Tiers?

Comment (mattiarighi) [reply esmvalcore/config-developer.yml:194]: > Do we really need those Tier1, 2, 3 directories? 
Yes, we should keep this classification, it's also described in the paper.

> Is it possible for one dataset to be in two different Tiers?
No, I don't think so.

I can also add another entry for dkrz with `Tier{tier}` and leave `default` unchanged.

Do we still need the `era5cli` entry? I think that would never work given the ambiguity in variable name.

Comment (bouweandela) [reply esmvalcore/config-developer.yml:194]: > I can also add another entry for dkrz with Tier{tier} and leave default unchanged.

Then we still need it in the recipes, so it would not help. Let's keep it then.

> Do we still need the era5cli entry? I think that would never work given the ambiguity in variable name.

I think it can be deleted, I guess I would need to add some documentation somewhere then on what directory structure to use.

**Comment by bouweandela** [comment]
I'll add some documentation after ESMValGroup/ESMValTool#1652 is merged.
